doc/mgmt/SONiC_YANG_Model_Guidelines.md

```diff
@@ -511,51 +512,206 @@ Example : "INTERFACE" table stores VRF names to which an interface belongs, also
+
+```yang
 ......
-container sonic-interface {
-	container INTERFACE {
-		list INTERFACE_LIST {  // 1st list
-    			key ifname;
-			
-			leaf ifname {
-				type leafref { 
-				......
-				}
+container INTERFACE {
+	list INTERFACE_LIST {  // 1st list
+		key ifname;
+
+		leaf ifname {
+			type leafref {
+			......
 			}
-			leaf vrf-name {
-				type leafref { 
-				......
-				}
+		}
+		leaf vrf-name {
+			type leafref {
+			......
 			}
-    			......
 		}
+		......
+	}
 
-		list INTERFACE_IPADDR_LIST { //2nd list
-   			key ifname, ip_addr;
-			
-  			leaf ifname {
-				type leafref { 
-				......
-				}
-			}
-  			leaf ip_addr {
-      				  type inet:ipv4-prefix;
-  			}
+	list INTERFACE_IPADDR_LIST { //2nd list
+		key "ifname ip_addr"
+
+		leaf ifname {
+			type leafref {
 			......
+			}
+		}
+		leaf ip_addr {
+			  type inet:ipv4-prefix;
+		}
+		......
+	}
+}
+......
+```
+***In the example above if the config DB contains an INTERFACE table with single key element then it will be associted with the INTERFACE_LIST and if contains 2 key elements then it will be associated with INTERFACE_IPADDR_LIST***
+
+#### Example 2: Keys with same number of elements of same type (NOT Allowed case 1)
+
+```yang
+......
+container NOT_SUPPORTED_INTERFACE {
+	list NOT_SUPPORTED_INTERFACE_LIST {  // 1st list
+		key ifname;
+		leaf ifname {
+			type string;
+		}
+		// ...
+	}
+
+	list NOT_SUPPORTED_INTERFACE_ANOTHER_LIST { // Negative case
+		key ifname;
+		leaf ifname {
+			type string;
 		}
-	} 
+		// ...
+	}
 } 
 ......
 ```
 
+***In the example above if the config DB contains an NOT_SUPPORTED_INTERFACE table with key Ethernet1 then it would match with both the list, this is an overlapping scenario***
+
+#### Example 3: Keys with same number of elements of same type (NOT Allowed case 2)
+
+```yang
+......
+container NOT_SUPPORTED_TELEMETRY_CLIENT {
+    list NOT_SUPPORTED_TELEMETRY_CLIENT_DS_LIST {  // 1st list
+        key "prefix name";
+
+        leaf prefix {
+            type string {
+                pattern "DestinationGroup_" + ".*";
+            }
+        }
+
+        leaf name {
+            type string;
+        }
+
+        leaf dst_addr {
+            type ipv4-port;
+        }
+    }
+
+    list NOT_SUPPORTED_TELEMETRY_CLIENT_SUB_LIST { // Negative case
+        key "prefix name";
+
+        leaf prefix {
+            type string {
+                pattern "Subscription_" + ".*";
+            }
+        }
+
+        leaf name {
+            type string;
+        }
+
+        leaf dst_group {
+            must "(contains(../../TELEMETRY_CLIENT_DS_LIST/prefix, current()))";
+            type string;
+        }
+    }
+}
+......
+```
+***In the example above if the config DB contains an NOT_SUPPORTED_TELEMETRY_CLIENT table with key "DestinationGroup|HS", then it would correspond to the NOT_SUPPORTED_TELEMETRY_CLIENT_DS_LIST and NOT_SUPPORTED_TELEMETRY_CLIENT_SUB_LIST, this is an overlapping scenario***
```

Comment:
Can we remove the ***bold italics*** formatting here and in other example descriptions? This text looks same as the "Example X" sub headings.